gravitee-apim-gateway/gravitee-apim-gateway-handlers/gravitee-apim-gateway-handlers-api/src/main/java/io/gravitee/gateway/reactive/handlers/api/processor/ApiProcessorChainFactory.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.gateway.reactive.handlers.api.processor;
+
+import io.gravitee.gateway.reactive.core.processor.Processor;
+import io.gravitee.gateway.reactive.core.processor.ProcessorChain;
+import io.gravitee.gateway.reactive.handlers.api.processor.cors.CorsPreflightRequestProcessor;
+import io.gravitee.gateway.reactive.handlers.api.processor.cors.CorsSimpleRequestProcessor;
+import io.gravitee.gateway.reactive.handlers.api.processor.forward.XForwardedPrefixProcessor;
+import io.gravitee.gateway.reactive.handlers.api.processor.pathmapping.PathMappingProcessor;
+import io.gravitee.gateway.reactive.handlers.api.processor.shutdown.ShutdownProcessor;
+import io.gravitee.node.api.Node;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * @author Guillaume LAMIRAND (guillaume.lamirand at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+public class ApiProcessorChainFactory {
+
+    private final Options options;
+    private final Node node;
+    private ProcessorChain preProcessorChain;
+    private ProcessorChain postProcessorChain;
+
+    public ApiProcessorChainFactory(final Options options, Node node) {
+        this.options = options;
+        this.node = node;
+    }
+
+    public ProcessorChain preProcessorChain() {
```

Comment:
Do you think we could have a factory that could take the api in parameter like we have here https://github.com/gravitee-io/gravitee-api-management/blob/prototype-reactive/gravitee-apim-gateway/gravitee-apim-gateway-handlers/gravitee-apim-gateway-handlers-api/src/main/java/io/gravitee/gateway/reactive/handlers/api/flow/resolver/FlowResolverFactory.java

Reply:
The processor chain are created for all api once and then shared. The API itself is retrieved when executing the processor. so in this case, no I don't want to create 1 processor chain by api to avoid many many instances of it.

Reply:
I will do as you explained on slack: one chain by api but we shared processors instances.